packages/tfchain_client/src/client.ts

```diff
@@ -0,0 +1,244 @@
+  static lock: AwaitLock = new AwaitLock();
+  keypair: KeyringPair;
+  address: string;
+  contracts: Contracts = new Contracts(this);
+  utility: Utility = new Utility(this);
+  balances: Balances = new Balances(this);
+  nodes: Nodes = new Nodes(this);
+  termsAndConditions: TermsAndConditions = new TermsAndConditions(this);
+  kvstore: KVStore = new KVStore(this);
+  twins: Twins = new Twins(this);
+  constructor(
+    public url: string,
+    public mnemonicOrSecret: string,
+    public keypairType: KeypairType = "sr25519",
+    public extSigner: ExtSigner | null = null,
+  ) {
+    super(url);
+    if (!SUPPORTED_KEYPAIR_TYPES.includes(keypairType)) {
+      throw Error(`Keypair type ${keypairType} is not a valid type. Should be either of: ${SUPPORTED_KEYPAIR_TYPES}`);
+    }
+    if (!url) {
+      throw Error("url should be provided");
+    }
+    if ((!mnemonicOrSecret && !extSigner) || (mnemonicOrSecret && extSigner)) {
+      throw Error("mnemonicOrSecret or extension signer should be provided");
+    }
+
+    if (mnemonicOrSecret && !validateMnemonic(this.mnemonicOrSecret)) {
+      throw Error("Invalid mnemonic! Must be bip39 compliant");
+    }
+  }
+
+  async loadKeyPairOrSigner(): Promise<void> {
+    if (this.mnemonicOrSecret && !this.address) {
+      const keyring = new Keyring({ type: this.keypairType });
+      await waitReady();
+
+      try {
+        this.keypair = keyring.addFromUri(this.mnemonicOrSecret);
+        this.address = this.keypair.address;
+      } catch (error) {
+        throw Error("Invalid mnemonic or secret seed! Please check your input.");
+      }
+    }
+    if (this.extSigner && !this.address) {
+      this.address = this.extSigner.address;
+    }
+  }
+
+  private async _applyExtrinsic<T>(
+    extrinsic: SubmittableExtrinsic<"promise", ISubmittableResult>,
+    resultSections: string[] = [""],
+  ): Promise<T> {
+    await this.connect();
+
+    const promise = new Promise(async (resolve, reject) => {
+      function callback(res) {
+        if (res instanceof Error) {
+          console.error(res);
+          reject(res);
+        }
+        const { events = [], status } = res;
+        const resultData: T[] = [];
+        if (!(extrinsic.method.section === "utility" && ["batch", "batchAll"].includes(extrinsic.method.method))) {
+          resultSections.push(extrinsic.method.section);
+        }
+        if (status.isInBlock) {
+          events.forEach(({ phase, event: { data, method, section } }) => {
+            console.log(`phase: ${phase}, section: ${section}, method: ${method}, data: ${data}`);
+            if (section === "system" && method === "ExtrinsicFailed") {
+              const errorIndex = parseInt(data.toJSON()[0].module.error.replace(/0+$/g, ""));
+              reject(errorIndex);
+            } else if (resultSections.includes(section)) {
+              resultData.push(data.toPrimitive()[0]);
+            } else if (section === "system" && method === "ExtrinsicSuccess") {
```

Comment:
The hard coded of the `ExtrinsicSuccess` here, can we put it inside an enum just like.

```suggestion
enum TFChainMethods {
  ExtrinsicFailed = 'ExtrinsicFailed',
  ExtrinsicSuccess = 'ExtrinsicSuccess',
}
} else if (section === "system" && method === TFChainMethods.ExtrinsicSuccess) {
```
if we changed it in the feature for any reason, we don't have to search for it in the whole project.
it's just a question, if you see something different please go ahead.